proto/cacheclient.proto

```diff
@@ -207,6 +207,9 @@ message _ListPushFrontRequest {
   bytes value = 2;
   uint64 ttl_milliseconds = 3;
   bool refresh_ttl = 4;
+
+  // ensure total length <= this; remove excess from back of list
+  optional uint64 truncate_tail_to_size = 5;
```

Comment:
we are using `uint32 ` for list length. Do you think we should update that?

Reply:
Can do that. I was looking at the usage of `uint64` for `ttl_milliseconds`.

Reply:
On second thoughts, I think `uint32` should suffice. The ttl is in milliseconds and that is why we used `uint64` there.